convert_to_hf.py

```diff
@@ -0,0 +1,102 @@
+import json
+import os
+import re
+
+import torch
+from tqdm.auto import trange
+from transformers import AutoConfig, PretrainedConfig
+
+
+def get_num_layers(config) -> int:
+    match config["model_type"]:
+        case "llama":
+            return config["num_hidden_layers"]
+        case unknown_type:
+            raise NotImplementedError(f"Can't get number of layers for {unknown_type}")
+
+
+def get_layers_prefix(config) -> str:
+    match config["model_type"]:
+        case "llama":
+            return "model.layers"
+        case unknown_type:
+            raise NotImplementedError(f"Can't get layers prefix for {unknown_type}")
+
+
+def pack_ints(data: torch.IntTensor, nbits: int) -> torch.IntTensor:
```

Comment:
The packing strategy handles negative codes for int32, but the kernel only checks for int16

Reply:
There is no more kernel

Reply:
I brought it back